### DISCUSSION TITLE
Parameter ''using: composite ' is not supported, use 'docker' or 'node12' instead. Github Server 3.0.1

### DISCUSSION THREAD

**Discussion by anthonydahanne** [community/community#24922] [Actions]
<p>Hello all,<br>
The title says it all, according to the Actions documentation, it should be possible to use <code>composite</code> actions, but when I tried using an action with such <code>action.yml</code>:</p>
<pre><code class="lang-auto">runs:
  using: 'composite '
  steps:
    - run: ${{ github.action_path }}/myscript ${{ inputs.my-input }}
      shell: bash
</code></pre>
<p>from a repository with such <code>.github/worflow/test.yml</code> :</p>
<pre><code class="lang-auto">    - uses: Actions/my-action-repo@main
      name: A name
      id: tests
      with:
        my-input: 'tests'
</code></pre>
<p>Are composite actions available to GH Server users ?<br>
Thank you</p>

**Comment by otter-computer** [comment]
<p>Hey <a class="mention" href="/anthonydahanne">@anthonydahanne</a>, that is strange. Composite actions <em>should</em> be supported on Enterprise installs so I don’t know why you’d be seeing that specific error. Would you be able to post your full workflow file and the full composite action so we can see what’s going on here and attempt to reproduce?</p>

**Comment by Simran-B** [comment, marked as answer]
<aside class="quote no-group" data-username="anthonydahanne" data-post="1" data-topic="167102">
<div class="title">
<div class="quote-controls"></div>
 anthonydahanne:</div>
<blockquote>
<p><code>  using: 'composite '</code></p>
</blockquote>
</aside>
<p>There’s a space between <code>e</code> and <code>'</code> which shouldn’t be there. Maybe that’s the problem?</p>

**Comment by anthonydahanne** [comment]
<p>Hello <a class="mention" href="/simran-b">@Simran-B</a> ,<br>
Good catch, indeed I got a trailing whitespace! and removing it, made it work.</p>
<p><a class="mention" href="/thomasshaped">@thomasshaped</a> thanks for the suggestion, I was about to give more detail, but <a class="mention" href="/simran-b">@Simran-B</a> suggestion was spot on!</p>
<p><a class="mention" href="/thomasshaped">@thomasshaped</a> the error message is not explicit though, I suggest it should have displayed:</p>
<pre><code class="lang-bash">Parameter ”using: composite ’ is not supported, use ‘docker’ or ‘node12’ or ‘composite’ instead.
</code></pre>
<p>instead of:</p>
<pre><code class="lang-bash">Parameter ”using: composite ’ is not supported, use ‘docker’ or ‘node12’ instead.
</code></pre>
<p>That would have hinted me that <code>composite</code> is well supported and I only made a typo in it - can you direct me to a bug /feature request?</p>
<p>Thanks to you both!</p>

**Comment by otter-computer** [comment]
<p>Omg what a find, I totally didn’t see that either :joy: Thanks, <a class="mention" href="/simran-b">@Simran-B</a>! And yeah, I’ll report this to the Actions team because that’s a confusing error message :v:</p>
